CMakeLists.txt

```diff
@@ -20,6 +20,19 @@ endif()
 
 project(transmission)
 
+# Emulates PROJECT_IS_TOPLEVEL (CMake 3.21)
+if ("${PROJECT_SOURCE_DIR}" STREQUAL "${CMAKE_SOURCE_DIR}")
+    if (("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}"))
+        message(FATAL_ERROR "In-tree builds are unsupported!")
+    endif()
+
+    # Write a .gitignore file in the binary dir to better manage VCS
+    if (NOT EXISTS "${CMAKE_BINARY_DIR}/.gitignore")
```

Comment:
Do we need to do this unconditionally?
Like, we have tarball buils, with no VCS present

Reply:
I don't think cmake has a way to tell when it's under VCS or not. The creation of this file is pretty low overhead, I can't imagine it being an issue.